java/vortex-jni/src/main/java/dev/vortex/api/expressions/proto/EndianUtils.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * (c) Copyright 2025 SpiralDB Inc. All rights reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.vortex.api.expressions.proto;
+
+import com.google.protobuf.ByteString;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+final class EndianUtils {
+    static byte[] reverse(ByteString src) {
+        byte[] dst = new byte[src.size()];
+        for (int i = 0; i < dst.length; i++) {
+            dst[i] = src.byteAt(dst.length - 1 - i);
+        }
+        return dst;
+    }
+
+    static byte[] littleEndianDecimal(BigDecimal decimal) {
+        BigInteger unscaled = decimal.unscaledValue();
+        byte[] bigEndianBytes = unscaled.toByteArray();
+
+        // Determine target size (1, 2, 4, 8, 16, or 32 bytes)
+        int targetSize;
+        if (bigEndianBytes.length <= 1) targetSize = 1;
+        else if (bigEndianBytes.length <= 2) targetSize = 2;
+        else if (bigEndianBytes.length <= 4) targetSize = 4;
+        else if (bigEndianBytes.length <= 8) targetSize = 8;
+        else if (bigEndianBytes.length <= 16) targetSize = 16;
+        else targetSize = 32;
```

Comment:
is it possible to be bigger than this...? should we be checking that `bigEndianBytes.length <= 32`?

Reply:
good point